resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/security/YARNHadoopDelegationTokenManager.scala

```diff
@@ -44,6 +44,10 @@ private[yarn] class YARNHadoopDelegationTokenManager(
 
   // public for testing
   val credentialProviders = getCredentialProviders
+  if (credentialProviders.keys.nonEmpty) {
```

Comment:
`credentialProviders.nonEmpty` should be enough?

Reply:
Yup.